pkg/validation/limits.go

```diff
@@ -184,6 +184,9 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 
 	_ = l.PerTenantOverridePeriod.Set("10s")
 	f.Var(&l.PerTenantOverridePeriod, "limits.per-user-override-period", "Period with this to reload the overrides.")
+
+	_ = l.QuerySplitDuration.Set("30m")
+	f.Var(&l.QuerySplitDuration, "limits.split-queries-by-interval", "Split queries by an interval and execute in parallel, 0 disables it. You should use an a multiple of 24 hours (same as the storage bucketing scheme), to avoid queriers downloading and processing the same chunks. This also determines how cache keys are chosen when result caching is enabled")
```

Comment:
why name this `limits.split-queries-by-interval` instead of `querier.split-queries-by-interval`? If we did that latter, this would be a non-breaking change for existing configs already setting `querier.split-queries-by-interval`. There is precedent for doing that already, like with `querier.max-streams-matcher-per-query`.

Reply:
+1 for this. We already alias old configs into the global limits struct many times, so let's keep the `querier.split-queries-by-interval` flag.

Reply:
Addressed in commit: https://github.com/grafana/loki/pull/5184/commits/d35fcf361824a3ac360a21c9d0e75803ad0c6b06